lib/hls/hls_parser.js

```diff
@@ -1695,6 +1695,22 @@ shaka.hls.HlsParser = class {
     const canSkipSegments = serverControlTag ?
           serverControlTag.getAttribute('CAN-SKIP-UNTIL') != null : false;
 
+    /** @type {!Array.<!shaka.hls.Tag>} */
+    const dateRangeTags = [];
+    if (playlist.segments) {
+      for (const segment of playlist.segments) {
+        const segmentDateRangeTags = shaka.hls.Utils.filterTagsByName(
+            segment.tags, 'EXT-X-DATERANGE');
+        dateRangeTags.push(...segmentDateRangeTags);
+      }
+    }
+
+    const syncTime = segmentIndex.earliestReference().syncTime;
```

Comment:
`syncTime` here is the first segment's PDT in epoch seconds?

why do we need to subtract it here, but in dash case the startTime returning doesn't need to subtract any offset? 
https://github.com/harmonicinc-com/shaka-player/blob/204e2a0fbb7aef7964e03d4dc8ad3652e826f20d/lib/dash/dash_parser.js#L1976

does syncTime increase as older segments are removed from playlist?

Reply:
In the HLS date range tag, the `START-DATE` is a datetime, which is different from the time values used in the `TimelineRegionInfo` (the startTime in the beginning is set to 0). In order for the observer to correctly be notified of events such as entering/exiting a region, the datetime has to be offset to reflect the video time.

Yes, the syncTime does increase when the playlist is updated. The previous implementation is incorrect. Updating it now to account for when segments are removed from the playlist.